### ISSUE TITLE
Update twisted.application.service.IProcess to use Attributes

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4534]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4534|
|Type|task|
|Created|2010-06-30 13:35:34Z|
|Branch|https://github.com/twisted/twisted/tree/iprocess-attributes-4534|

`IProcess` defines its attribute in the old style, of epytext markup on the interface docstring.  It should use the new style, of `zope.interface.Attribute`.

This would let `verifyClass` and `verifyObject` be more useful in conjunction with it.

Specifically this is motivated by a desire to use `verifyObject` in `twisted.test.test_twistd` against one of the fakes used.

<details><summary>Searchable metadata</summary>

```
trac-id__4534 4534
type__task task
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_iprocess_attributes_4534 branches/iprocess-attributes-4534
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1277904934000000 1277904934000000
changetime__1279029479000000 1279029479000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [29497]) Branching to 'iprocess-attributes-4534'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

[build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/iprocess-attributes-4534)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Please merge!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [29798]) Merge iprocess-attributes-4534

Author: exarkun
Reviewer: therve
Fixes: [#4534](https://github.com/twisted/twisted/issues/4534)

Change the definition of `twisted.application.service.IProcess` to use
`zope.interface.Attribute` to specify its attributes, rather than epytext
in the interface docstring.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
